### PR TITLE
Avoid stream errors for short-lived commands

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

This avoids stream errors on short-lived commands like `vast -N dump concepts`. These were really non-errors caused by early exits from shutting down the node. We shouldn't print red stuff when there isn't actually an error.

> 2020-11-26T14:40:06.085 archive got a stream error: exit_reason(user_shutdown)
> 2020-11-26T14:40:06.086 index aborts with error: !! exit

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Run locally before and after the change I've made, verify that the quoted errors are no more.

### CODE REVIEW

File: libvast/src/system/archive.cpp [hunk 1]
```diff
@@ -185,9 +185,17 @@ archive(archive_type::stateful_pointer<archive_state> self, path dir,
           t.stop(events);
         },
         [=](unit_t&, const error& err) {
-          if (err) {
-            VAST_ERROR(self, "got a stream error:", self->system().render(err));
+          // We get an 'unreachable' error when the stream becomes unreachable
+          // because the actor was destroyed; in this case we can't use `self`
+          // anymore.
```
Comment (lava) [inline libvast/src/system/archive.cpp:190]: In retrospect I'm not sure anymore that the comment is 100% accurate: Removing the check in the partition gives occasional segfaults due to null pointer dereference of `self`, but a trivial reproduction of the situation did not have the same issue, so maybe this is something specific to the setup of the index/partition. (i was planning to investigate this for the streaming blog post as well, but didn't really get to it yet)

File: libvast/src/system/archive.cpp [hunk 2]
```diff
@@ -185,9 +185,17 @@ archive(archive_type::stateful_pointer<archive_state> self, path dir,
...
+          if (err && err != caf::exit_reason::unreachable) {
+            if (err != caf::exit_reason::user_shutdown)
```
Comment (lava) [inline libvast/src/system/archive.cpp:192]: I think `user_shutdown` might also be used as a default argument when shutting down the stream, not just when its literally the user shutting down the stream, but I don't completely remember tbh. Did you search for this in the CAF source code?

Comment (dominiklohmann) [reply libvast/src/system/archive.cpp:192]: Yes, user_shutdown is just the default exit reason. It doesn't make sense to treat it as an error.

File: libvast/src/system/archive.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/src/system/archive.cpp:190]: Should we investigate this separately from this PR?

File: libvast/src/system/archive.cpp [hunk 3]
```diff
@@ -185,9 +185,17 @@ archive(archive_type::stateful_pointer<archive_state> self, path dir,
...
+              VAST_ERROR(self, "got a stream error:", render(err));
```
Comment (lava) [inline libvast/src/system/archive.cpp:193]: I think we should still print the error on `DEBUG`, to be able to distinguish between unreachable and user_shutdown in the logs.

**Review by lava — APPROVED** [review]
Looking good

**Pull request merged**